app/scripts/modules/kubernetes/src/v2/pipelines/stages/runJob/KubernetesV2RunJobStageConfig.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+
+import {
+  IStageConfigProps,
+  AccountService,
+  //   HelpField,
+  YamlEditor,
+  yamlDocumentsToString,
+  IAccount,
+} from '@spinnaker/core';
+
+import { ManifestBasicSettings } from 'kubernetes/v2/manifest/wizard/BasicSettings';
+
+export interface IKubernetesRunJobStageConfigState {
+  credentials: IAccount[];
+  account: string;
+  rawManifest?: string;
+}
+
+export class KubernetesV2RunJobStageConfig extends React.Component<IStageConfigProps> {
+  public state: IKubernetesRunJobStageConfigState = {
+    credentials: [],
+    account: '',
+  };
+
+  public constructor(props: IStageConfigProps) {
+    super(props);
+    const { stage } = props;
+    stage.cloudProvider = stage.cloudProvider || 'kubernetes';
```

Comment:
If we're updating the stage let's do it in `componentDidMount`. Let's also use `updateStageField` rather than mutating the stage and then calling `stageFieldUpdated`.